### DISCUSSION TITLE
How many Chaos Mesh pods should I expect in a 4-node k8s cluster?

### DISCUSSION THREAD

**Discussion by bruteMax** [chaos-mesh/chaos-mesh#3175] [Q&A]
Hello.  I have a 4-node cluster and have installed v2.1.5 via Helm.  This is what i see:
```
[root@k8s-node1 ~]#  kubectl get pods --namespace chaos-mesh
NAME                                        READY   STATUS    RESTARTS   AGE
chaos-controller-manager-78777c8bc8-b4hvg   1/1     Running   0          26m
chaos-controller-manager-78777c8bc8-t66hc   1/1     Running   0          26m
chaos-controller-manager-78777c8bc8-wdkz9   1/1     Running   0          26m
chaos-daemon-bzjlg                          1/1     Running   0          26m
chaos-daemon-fpjnc                          1/1     Running   0          26m
chaos-daemon-l6s8x                          1/1     Running   0          26m
chaos-daemon-z2xz7                          1/1     Running   0          26m
chaos-dashboard-5bd55ff6b8-2ls75            1/1     Running   0          26m
```
I assume there is a daemon pod for each k8s node, so why only 3 controller managers pods?

Thank you.

**Comment by STRRL** [comment]
Hi @bruteMax, the "daemon Pod" is `chaos-daemon-xxxxx`, you could find that there are 4 pods here. :)

**Reply by bruteMax** [reply]
Thank you, but I am mainly wondering why there are only 3 controller-manager pods?  Should there not be 4 as well?

**Reply by nioshield** [reply]
controller-manager is CRD Controller,3 Controller-manager for  high availability

**Comment by xlgao-zju** [comment]
there are three types of pods in a chaos mesh:

- 3 chaos-controller-manager pods to make up a high available controller cluster. 
- 1 dashboard pod.  
- the sum of chaos-daemon will be the same as the sum of your nodes in k8s, since this pods is generated by a `daemonSet`.
